clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -87,7 +91,32 @@ public SslFactory(Mode mode, String clientAuthConfigOverride, boolean keystoreVe
     @Override
     public void configure(Map<String, ?> configs) throws KafkaException {
         this.protocol =  (String) configs.get(SslConfigs.SSL_PROTOCOL_CONFIG);
-        this.provider = (String) configs.get(SslConfigs.SSL_PROVIDER_CONFIG);
+        String provider = (String) configs.get(SslConfigs.SSL_PROVIDER_CONFIG);
```

Comment:
we can make this more similar to other broker pluggability mechanisms: make this config SSL_PROVIDER_CLASSNAME, then just do something like `this.sslContextProvider = (Supplier<SSLContext>) Class.forName((String) configs.get(SslConfigs.SSL_PROVIDER_CLASSNAME_CONFIG)).newInstance()`

bonus points if you make an interface for this (beyond just Supplier) and make it implement Configurable too (see  `authorizerClassName` handling in broker code for example)

Reply:
I feel it is a little lengthy that way, also there is new config exposed to user. 
I have tried to refactor the code that way. PTAL at the latest commit.